app/src/pages/playground/playgroundUtils.ts

```diff
@@ -1043,3 +1058,98 @@ export function normalizeMessageAttributeValue(
     ? content
     : JSON.stringify(content, null, 2);
 }
+
+export function areRequiredInvocationParametersConfigured(
+  configuredInvocationParameters: InvocationParameterInput[],
+  supportedInvocationParameters: InvocationParameter[]
+) {
+  return supportedInvocationParameters
+    .filter((param) => param.required)
+    .every((param) =>
+      configuredInvocationParameters.some((ip) =>
+        areInvocationParamsEqual(ip, param)
+      )
+    );
+}
+
+/**
+ * Extracts the default value for the invocation parameter definition
+ * And the key name that should be used in the invocation parameter input if we need to make a new one
+ *
+ * This logic is necessary because the default value is mapped to different key name based on its type
+ * within the InvocationParameterInput queries in the playground
+ */
+const getInvocationParamDefaultValue = (
+  param: InvocationParameter
+): {
+  invocationInputField: string | undefined;
+  defaultValue: unknown;
+} => {
+  let defaultValueKeyName: string | undefined;
+  Object.entries(param).find(([key, value]) => {
+    if (key.endsWith("DefaultValue") && value != null) {
+      defaultValueKeyName = key;
+      return true;
+    }
+    return false;
+  });
+  return {
+    invocationInputField: param.invocationInputField,
+    defaultValue:
+      defaultValueKeyName != null
+        ? param[defaultValueKeyName as keyof InvocationParameter]
+        : undefined,
+  };
+};
+
+/**
+ * Merges the current invocation parameters with the default values for the supported invocation parameters,
+ * only adding values for invocation parameters that don't already have a value
+ */
+export function mergeInvocationParametersWithDefaults(
+  invocationParameters: InvocationParameterInput[],
+  supportedInvocationParameters: InvocationParameter[]
+) {
+  // Convert the current invocation parameters to a map for quick lookup
+  const currentInvocationParametersMap = new Map(
+    invocationParameters.map((param) => [
+      param.canonicalName || param.invocationName,
+      param,
+    ])
+  );
+  supportedInvocationParameters.forEach((param) => {
+    const paramKeyName = param.canonicalName || param.invocationName;
+    // Extract the default value for the invocation parameter definition
+    // And the key name that should be used in the invocation parameter input if we need to make a new one
+    const { invocationInputField, defaultValue } =
+      getInvocationParamDefaultValue(param);
+    // Skip if we don't have required fields
+    // or, if the current invocation parameter map already has a value for the key
+    // so that we don't overwrite a user provided value, or a value saved to preferences
+    if (
+      !param.invocationName ||
+      !paramKeyName ||
+      defaultValue == null ||
+      currentInvocationParametersMap.get(paramKeyName)?.[
+        toCamelCase(
+          param.invocationInputField || ""
+        ) as keyof InvocationParameterInput
+      ] != null ||
```

Comment:
we should pull this out to make it a bit easier to read. The toCamelCase is because all params are stored like that in the map? when would these things be out of sync? one needing to be camelCased cause the other already had been

Reply:
Good callout, done